js/ding_availability.js

```diff
@@ -130,6 +130,9 @@
           }
         }
       }
+      else {
+        reserver_btn.addClass('not-reservable');
+      }
```

Comment:
Why add the `not-reservable` class to an element which we do not know the status for?

If we are to add a class I would expect it that we should add it to the element as well as in the other branches of the code.